shared/js/pageStore.js

```diff
@@ -63,8 +71,15 @@ loop.store.PageStore = function(mozL10n) {
 
     /**
      * Handle AddedPage action by updating the store state.
+     *
+     * @note If actionData.url already exists in the store, it will not
+     * be added a second time.
      */
     addedPage(actionData) {
+      if (this._hasPageForUrl(actionData.url)) {
+        return;
+      }
```

Comment:
Do we really need this second check here? I mean, we don't allow to save duplicate content, so I can't see a case where an `AddedPage` action will throw with duplicate url.